src/components/Calculator.js

```diff
@@ -1,43 +1,71 @@
 import React from 'react';
+import calculate from '../logic/calculate';
+import Buttons from './buttons';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      calcObj: {},
+    };
+
+    this.handleCalculate = this.handleCalculate.bind(this);
+  }
+
+  handleCalculate(calcObj, text) {
+    this.setState({ calcObj: calculate(calcObj, text) });
   }
 
   render() {
+    const { calcObj } = this.state;
+
+    let calcDisplay = '0';
+    if (
+      Object.keys(calcObj).length === 0
+      || (calcObj.total === null
+        && calcObj.next === null
+        && calcObj.operation === null)
+    ) {
+      calcDisplay = '0';
+    } else {
+      calcDisplay = calcObj.next != null ? calcObj.next : calcObj.total;
+    }
```

Comment:
[OPTIONAL]

- The calculator is working fine and I appreciate the work that you have done. Here I came with one suggestion to make the UI much better. Currently, I can't see the `operator` like `+`, `-`, etc.. on the display screen when I pressed, it's better if you work on that to give a better user experience.